cpp/oneapi/dal/table/detail/csr_accessor.hpp

```diff
@@ -0,0 +1,53 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/table/detail/table_impl_iface.hpp"
+
+namespace oneapi::dal::detail {
+namespace v1 {
+
+/// @tparam T The type of data values in blocks returned by the accessor.
+///           Should be const-qualified for read-only access.
+///           An accessor supports at least :expr:`float`, :expr:`double`, and :expr:`std::int32_t` types of :literal:`T`.
+template <typename T>
+class csr_accessor {
+public:
+    using data_t = std::remove_const_t<T>;
+    static constexpr bool is_readonly = std::is_const_v<T>;
+
+    /// Creates a new read-only accessor object from the csr table.
+    template <typename K, typename = std::enable_if_t<is_readonly>>
+    csr_accessor(const K& obj) {
+        csr_access_iface_ = &cast_impl<sparse_table_impl_iface>(obj);
+    }
+
+    sparse_block<data_t> pull(const range& rows = { 0, -1 }) const {
+        sparse_block<data_t> block;
+        csr_access_iface_->pull(block, rows);
+        return block;
+    }
+
```

Comment:
Can you please declare all the method required for accessors here including DPC++ case, but throw an unimplemneted exception inside?